### PR TITLE
Add find_max_in_list function with tests

### PR DESCRIPTION
**Pull request by yuri-spizhovyi-mit**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [ ] The file name describes the function's behavior
- [ ] There is a module docstring in the function file
- [ ] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [ ] There is a module docstring in the tests file

### Unit Tests

- [ ] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Every unit test has
  - [ ] A helpful name
  - [ ] A clear docstring
  - [ ] Only one assertion
  - [ ] There is no logic in the unit test
- [ ] All tests pass
- [ ] There are tests for defensive assertions
- [ ] There are tests for boundary cases

### Function Docstring

- [ ] The function's behavior is described
- The function's arguments are described:
  - [ ] Type
  - [ ] Purpose
  - [ ] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [ ] Type
  - [ ] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [ ] Each assumption about an argument is checked with an assertion
  - [ ] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests

### The Function

- [ ] The function's name describes it's behavior
- [ ] The function's name matches the file name
- [ ] The function has correct type annotations

### Do's

- [ ] Variable names help to understand the strategy
- [ ] Any comments are clear and describe the strategy
- [ ] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [ ] The function's strategy _is not_ described in the documentation
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [ ] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [ ] The code passes the formatting checks
- [ ] The code passes all Ruff linting checks
- [ ] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [ ] Variables are named with snake_case
- [ ] Variable names are clear and helpful
- [ ] The code follows the strategy as simply as possible
- [ ] The implementation is as simple as possible given the strategy
- [ ] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible

### CODE REVIEW

**Review by Manezhahm — APPROVED** [review]

**Review by Manezhahm — COMMENTED** [review]
tut

**Comment by Manezhahm** [comment]
U

File: solutions/find_max_in_list.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on 2024-12-27
+
```
Comment (olumide-AI) [inline solutions/find_max_in_list.py:5]: You could write a better test module 

example
"""
Test module for count_between function.
Contains intentionally buggy tests for debugging practice.

Test categories:
    - Standard cases: typical number lists and ranges
    - Edge cases: empty lists, equal bounds, non-integer numbers
    - Defensive tests: invalid inputs, assertions
"""

File: solutions/tests/test_find_max_in_list.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Unit tests for the find_max_in_list function.
+
```
Comment (olumide-AI) [inline solutions/tests/test_find_max_in_list.py:5]: You could write a better module, 
example:

"""
Test module for count_between function.
Contains intentionally buggy tests for debugging practice.

Test categories:
    - Standard cases: typical number lists and ranges
    - Edge cases: empty lists, equal bounds, non-integer numbers
    - Defensive tests: invalid inputs, assertions
"""

File: solutions/find_max_in_list.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    >>> print(find_max_in_list([]))
+    None
+
+    >>> print(find_max_in_list([0, 0]))
+    0
+
+    >>> print(find_max_in_list([1, 2, 3]))
+    3
+
```
Comment (olumide-AI) [inline solutions/find_max_in_list.py:34]: You have too many doc string, evan said we should have at most 3-4

File: solutions/find_max_in_list.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    >>> find_max_in_list("not a list")
+    Traceback (most recent call last):
+    TypeError: Input must be a list of integers
+
+    >>> find_max_in_list([1, "2", 3])
+    Traceback (most recent call last):
+    TypeError: All elements in the list must be integers
+    """
+    if not isinstance(numbers, list):
```
Comment (olumide-AI) [inline solutions/find_max_in_list.py:43]: More line comments can be helpful to describe why the code is there and how it relates to other lines of the code. I think you have only one.

File: solutions/find_max_in_list.py (hunk 1 above)
Comment (yuri-spizhovyi-mit) [reply solutions/find_max_in_list.py:5]: Hi Olumide,
There's no need to duplicate the function description in this docstring. Please refer to the checklist for guidance.

File: solutions/find_max_in_list.py (hunk 3 above)
Comment (yuri-spizhovyi-mit) [reply solutions/find_max_in_list.py:34]: but I have 5😊 I followed the example given by Evan in the even_odd task (TDD with LLM lesson), where he created 7 doctests. I hope this aligns with the intent of the guidelines. 
Additionally, there are no strict rules about the number of tests—we generally aim for as many tests as necessary to ensure robust functionality. The more tests we have, the better we can cover different scenarios and edge cases. 
Let me know if you'd like to discuss further!

File: solutions/find_max_in_list.py (hunk 4 above)
Comment (yuri-spizhovyi-mit) [reply solutions/find_max_in_list.py:43]: Thank you for your feedback! If I understood your question correctly, the lines:
```
Traceback (most recent call last):  
TypeError: All elements in the list must be integers
```
are not comments; they are part of the doctests. These lines simulate the expected output of the code when a TypeError is raised, as required for proper test coverage. Let me know if I misunderstood or if you have additional suggestions!

**Comment by olumide-AI** [comment]
Yuriy code and function looks good, followed proper documentations and the code review checklist
